### PR TITLE
Common Data Binding: Add the Update Data article

### PR DESCRIPTION
**Pull request by RomanTsukanov**

### CODE REVIEW

File: concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/02 Data Shaping.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
+Data shaping is implemented by the [DataSource](/api-reference/30%20Data%20Layer/DataSource '/Documentation/ApiReference/Data_Layer/DataSource/') component and its methods. To call them, get a **DataSource** instance from your widget using the [getDataSource()](/api-reference/10%20UI%20Widgets/DataHelperMixin/3%20Methods/getDataSource().md '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Methods/#getDataSource') method. Alternatively, you can use a standalone instance saved in a constant/component property when you created the **DataSource**.
+
+In the following code, we get a **DataSource** instance using both approaches and call one of the data shaping methods&mdash;[filter(filterExpr)](/api-reference/30%20Data%20Layer/DataSource/3%20Methods/filter(filterExpr).md '/Documentation/ApiReference/Data_Layer/DataSource/Methods/#filterfilterExpr'). Such methods only set up data shaping settings. To apply them, we call the [load()](/api-reference/30%20Data%20Layer/DataSource/3%20Methods/load().md '/Documentation/ApiReference/Data_Layer/DataSource/Methods/#load') method.
```
Comment (AlexJameson) [inline concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/02 Data Shaping.md:3]: >Such methods only set up data shaping settings.

Shouldn't it be 'set data shaping settings'?

Comment (artem-kurchenko) [reply concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/02 Data Shaping.md]: I would suggest adding a code that processes the result

  dataSource.load().then((items)=>{
        // process data here
  })

File: concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/10 Data Modification.md [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+            dataSource.load();
+        }
+
+        // ...
+    }
+    export default App;
+
+---
+
+If a component is [integrated with a push service](/Documentation/Guide/Data_Binding/Data_Layer/#Data_Modification/Integration_with_Push_Services), use the [push(changes)](/api-reference/30%20Data%20Layer/Store/3%20Methods/push(changes).md '/Documentation/ApiReference/Data_Layer/ArrayStore/Methods/#pushchanges') method for all operations instead. You can also use it to edit local data. This method accepts an array, so you can make multiple changes in one method call.
```
Comment (artem-kurchenko) [inline concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/10 Data Modification.md]: It cannot be `instead`. You can use push to make notifications in widgets but not for data updates if remote data is used.

Comment (artem-kurchenko) [reply concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/10 Data Modification.md]: Also, for me, it's still better to keep Push API separate like we have now
https://js.devexpress.com/Documentation/Guide/Data_Layer/Data_Layer/#Data_Modification/Integration_with_Push_Services
Or it should be referenced as more detailed explanation in the second part of this topic

**Review by AlexJameson — APPROVED** [review]

File: concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/02 Data Shaping.md (hunk 1 above)
Comment (RomanTsukanov) [reply concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/02 Data Shaping.md:3]: I'm not sure. Let's leave it to editors.

File: concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/02 Data Shaping.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
+Data shaping is implemented by the [DataSource](/api-reference/30%20Data%20Layer/DataSource '/Documentation/ApiReference/Data_Layer/DataSource/') component and its methods. To call them, get a **DataSource** instance from your widget using the [getDataSource()](/api-reference/10%20UI%20Widgets/DataHelperMixin/3%20Methods/getDataSource().md '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Methods/#getDataSource') method. Alternatively, you can use a standalone instance saved in a constant/component property when you created the **DataSource**.
```
Comment (dirk-pieterse) [inline concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/02 Data Shaping.md:1]:
```suggestion
The [DataSource](/api-reference/30%20Data%20Layer/DataSource '/Documentation/ApiReference/Data_Layer/DataSource/') component and its methods implement data shaping. To call these components, use the [getDataSource()](/api-reference/10%20UI%20Widgets/DataHelperMixin/3%20Methods/getDataSource().md '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Methods/#getDataSource') method to get a **DataSource** instance from your widget . You can also use a standalone instance saved in a constant/component property when you created the **DataSource**.
```

File: concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/02 Data Shaping.md (hunk 1 above)
Comment (dirk-pieterse) [inline concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/02 Data Shaping.md:3]:
```suggestion
In the following code, we use both approaches to get a **DataSource** instance and call one of the data shaping methods&mdash;[filter(filterExpr)](/api-reference/30%20Data%20Layer/DataSource/3%20Methods/filter(filterExpr).md '/Documentation/ApiReference/Data_Layer/DataSource/Methods/#filterfilterExpr'). These methods only specify data shaping settings. To apply them, we call the [load()](/api-reference/30%20Data%20Layer/DataSource/3%20Methods/load().md '/Documentation/ApiReference/Data_Layer/DataSource/Methods/#load') method.
```

Comment (dirk-pieterse) [reply concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/02 Data Shaping.md:3]: "To apply them" What does 'them' refer to here?

File: concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/10 Data Modification.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
+Data modification is implemented by the store and its methods. To call them, you need a store instance that you can get using the **DataSource**'s [store()](/api-reference/30%20Data%20Layer/DataSource/3%20Methods/store().md '/Documentation/ApiReference/Data_Layer/DataSource/Methods/#store') method.
```
Comment (dirk-pieterse) [inline concepts/70 Data Binding/03 Update Data/05 DevExtreme DataSource/10 Data Modification.md:1]:
```suggestion
Data modification is implemented by the store and its methods. To call these methods, use the **DataSource**'s [store()](/api-reference/30%20Data%20Layer/DataSource/3%20Methods/store().md '/Documentation/ApiReference/Data_Layer/DataSource/Methods/#store') method to get a store instance.
```

File: concepts/70 Data Binding/03 Update Data/20 Local Array/02 Angular.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+Ensure that the **dataSource** option is bound to the array using one- or two-way binding. Then, change the array using <a href="http://www.w3schools.com/js/js_array_methods.asp" target="_blank">standard methods</a>, and the widget will be updated automatically.
```
Comment (dirk-pieterse) [inline concepts/70 Data Binding/03 Update Data/20 Local Array/02 Angular.md]:
```suggestion
Ensure that one- or two-way binding is used to bind the **dataSource** option to the array. Then, use <a href="http://www.w3schools.com/js/js_array_methods.asp" target="_blank">standard methods</a> to change the array. This updates the widget automatically.
```

File: concepts/70 Data Binding/03 Update Data/20 Local Array/05 Vue.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
+Ensure that the **dataSource** option is bound to the array using one- or two-way binding. Then, change the array using <a href="http://www.w3schools.com/js/js_array_methods.asp" target="_blank">standard methods</a>, and the widget will be updated automatically.
```
Comment (dirk-pieterse) [inline concepts/70 Data Binding/03 Update Data/20 Local Array/05 Vue.md]:
```suggestion
Ensure that one- or two-way binding is used to bind the **dataSource** option to the array. Then, use <a href="http://www.w3schools.com/js/js_array_methods.asp" target="_blank">standard methods</a> to change the array. This updates the widget automatically.
```

File: concepts/70 Data Binding/03 Update Data/20 Local Array/08 React.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+Store the array in the state and pass it to the **dataSource** option. When you need to modify the array, create a new array and save it in the state using `this.setState`. Do not use standard array modification methods, like `push()` or `pop()`. They modify the original array, which is against the <a href="https://reactjs.org/docs/state-and-lifecycle.html#using-state-correctly" target="_blank">rules of updating React state</a>.
```
Comment (dirk-pieterse) [inline concepts/70 Data Binding/03 Update Data/20 Local Array/08 React.md]:
```suggestion
Store the array in the state and pass it to the **dataSource** option. When you need to modify the array, create a new array and use `this.setState` to save it in the state. Do not use standard array modification methods, like `push()` or `pop()`. They modify the original array, which is not according to the <a href="https://reactjs.org/docs/state-and-lifecycle.html#using-state-correctly" target="_blank">rules of updating a React state</a>.
```

File: concepts/70 Data Binding/03 Update Data/20 Local Array/15 jQuery.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+Change the array using <a href="http://www.w3schools.com/js/js_array_methods.asp" target="_blank">standard methods</a> and reassign it to the **dataSource** option using the [option(optionName, optionValue)](/api-reference/10%20UI%20Widgets/Component/3%20Methods/option(optionName_optionValue).md '/Documentation/ApiReference/Data_Visualization_Widgets/dxChart/Methods/#optionoptionName_optionValue') method.
```
Comment (dirk-pieterse) [inline concepts/70 Data Binding/03 Update Data/20 Local Array/15 jQuery.md:1]:
```suggestion
Use <a href="http://www.w3schools.com/js/js_array_methods.asp" target="_blank">standard methods</a> to change the array and the [option(optionName, optionValue)](/api-reference/10%20UI%20Widgets/Component/3%20Methods/option(optionName_optionValue).md '/Documentation/ApiReference/Data_Visualization_Widgets/dxChart/Methods/#optionoptionName_optionValue') method to reassign it to the **dataSource** option.
```

**Review by dirk-pieterse — APPROVED** [review]

**Pull request merged**